provider/terraform/website/docs/version_2_upgrade.html.markdown

```diff
@@ -0,0 +1,176 @@
+---
+layout: "google"
+page_title: "Terraform Google Provider Version 2 Upgrade Guide"
+sidebar_current: "docs-google-provider-version-2-upgrade"
+description: |-
+  Terraform Google Provider Version 2 Upgrade Guide
+---
+
+# Terraform Google Provider Version 2 Upgrade Guide
+
+~> **NOTE:** This upgrade guide is a work in progress and will not be completed until the release of version 2.0.0 of the provider later this year. Many of the topics discussed, except for the actual provider upgrade, can be performed using the most recent 1.X version of the provider.
+
+Version 2.0.0 of the Google provider for Terraform is a major release and includes some changes that you will need to consider when upgrading. This guide is intended to help with that process and focuses only on changes from version 1.X to version 2.0.0.
+
+Most of the changes outlined in this guide have been previously marked as deprecated in the Terraform plan/apply output throughout previous provider releases. These changes, such as deprecation notices, can always be found in the [Terraform Google Provider CHANGELOG](https://github.com/terraform-providers/terraform-provider-google/blob/master/CHANGELOG.md).
```

Comment:
Maybe call out `1.19.0`/`1.19.1` explicitly? To make it super clear that going `1.19.X` -> `2.0.0` is **pretty safe** but `1.18.X` -> `2.0.0` is maybe not.